src/jobs/terraform_deploy.yml

```diff
@@ -52,14 +52,31 @@ steps:
       echo "[credential]" >> ~/.gitconfig
       echo "  helper = store" >> ~/.gitconfig
   # Real work
-  - run: terraform workspace new <<parameters.tf-workspace>>
-  - run: terraform init -upgrade <<parameters.main-dir>>
+  - run: echo 'export TF_WORKSPACE=<<parameters.tf-workspace>>' >> $BASH_ENV
+  - run: |
+      cd <<parameters.main-dir>>
+      export TF_WORKSPACE=<<parameters.tf-workspace>>
+      terraform init -upgrade
   - when:
-      condition: <<parameters.var-file>>
+      condition: <<parameters.main-dir>>/<<parameters.var-file>>
       steps:
-        - run: terraform apply -input=false -auto-approve -var-file=<<parameters.var-file>> <<parameters.main-dir>> 
+        - run: |
+            cd <<parameters.main-dir>>
+            export TF_WORKSPACE=<<parameters.tf-workspace>>
+            terraform apply -input=false -auto-approve -var-file=<<parameters.var-file>> 
   - unless:
-      condition: <<parameters.var-file>>
+      condition: <<parameters.main-dir>>/<<parameters.var-file>>
       steps:
-        - run: terraform apply -input=false -auto-approve <<parameters.main-dir>>
- 
+        - run: |
+            cd <<parameters.main-dir>>
+            export TF_WORKSPACE=<<parameters.tf-workspace>>
```

Comment:
Can the exports be done as "environment" vars in the yaml?
https://circleci.com/docs/2.0/env-vars/#setting-an-environment-variable-in-a-step

Reply:
Sure.  I think that's conceivable

Reply:
Code has been updated to use the environment keyword.  Tested and working.

Reply:
Sweet!!!  Nice work!!!